sematic/ui/packages/main/src/pipelines/pod_lifecycle/PodLifecycleEvent.tsx

```diff
@@ -0,0 +1,78 @@
+import TimelineConnector from "@mui/lab/TimelineConnector/TimelineConnector";
+import TimelineContent from "@mui/lab/TimelineContent/TimelineContent";
+import TimelineDot from "@mui/lab/TimelineDot/TimelineDot";
+import TimelineItem from "@mui/lab/TimelineItem/TimelineItem";
+import TimelineSeparator from "@mui/lab/TimelineSeparator/TimelineSeparator";
+import Chip from "@mui/material/Chip/Chip";
+import Typography from "@mui/material/Typography/Typography";
+import { styled } from "@mui/system";
+import { Job } from "@sematic/common/lib/src/Models";
+import { format } from 'date-fns';
+import { useMemo } from "react";
+
+const TERMINATE_STATE: string = 'Deleted';
+const SpacedText = styled(Typography)`
+    margin: 4px 0;
+`;
+
+function StyledChip(props: React.ComponentProps<typeof Chip>) {
+    return <Chip {...props} sx={{ mr: 1 }} />
+};
+
+function StyledChipWithColor(props:
+    React.ComponentProps<typeof StyledChip> & { resourceState: string }) {
+    const { resourceState, ...restProps } = props;
+
+    const color = useMemo(() => getColorByState(resourceState), [resourceState]);
+
+    return <StyledChip {...restProps} color={color} label={resourceState} />
+}
+
+interface PodLifecycleEventProps {
+    podStatus: Job['status_history_serialization'][number];
+    isLast: boolean;
+    isFirst: boolean;
+}
+
+function getColorByState(podEvent: string) {
+    if (["Running", "Pending"].includes(podEvent)) {
+        return 'primary';
+    }
+    if (TERMINATE_STATE === podEvent) {
+        return 'success';
+    }
+    if (["Requested"].includes(podEvent)) {
```

Comment:
Yes. what you mentioned makes sense from a code logic point of view. But having that documents how each state is handled explicitly. So it might be more maintainable and avoids potential hidden bugs. 

Either way is fine with me. I removed it for now.